application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/config/connection/AgentConnectionConfig.java

```diff
@@ -0,0 +1,315 @@
+    private RackAware myRackAware = new RackAware();
+    private HostAware myHostAware = new HostAware();
+
+    public AgentConnectionConfig()
+    {
+
+    }
+
+    @JsonCreator
+    public AgentConnectionConfig(
+        @JsonProperty("enabled") final boolean enableAgent)
+    {
+        enabled = enableAgent;
+    }
+
+    @JsonProperty("enabled")
+    public boolean isEnabled()
+    {
+        return enabled;
+    }
+
+    @JsonProperty("enabled")
+    public void setEnabled(final boolean enableAgent)
+    {
+        enabled = enableAgent;
+    }
+
+    @JsonProperty("type")
+    public ConnectionType getType()
+    {
+        return myType;
+    }
+
+    @JsonProperty("type")
+    public void setType(final String type)
+    {
+        myType = ConnectionType.valueOf(type);
+    }
+
+    @JsonProperty("contactPoints")
+    public Map<String, Host> getContactPoints()
+    {
+        return myContactPoints;
+    }
+
+    @JsonProperty("contactPoints")
+    public void setMyContactPoints(final List<Host> contactPoints)
+    {
+        if (contactPoints != null)
+        {
+            myContactPoints = contactPoints.stream().collect(
+                    Collectors.toMap(Host::getHost, ks -> ks));
+        }
+    }
+
+    @JsonProperty("datacenterAware")
+    public void setDatacenterAware(final DatacenterAware datacenterAware)
+    {
+        myDatacenterAware = datacenterAware;
+    }
+
+    @JsonProperty("datacenterAware")
+    public DatacenterAware getDatacenterAware()
+    {
+        return myDatacenterAware;
+    }
+
+    @JsonProperty("rackAware")
+    public void setRackAware(final RackAware rackAware)
+    {
+        myRackAware = rackAware;
+    }
+
+    @JsonProperty("rackAware")
+    public RackAware getRackAware()
+    {
+        return myRackAware;
+    }
+
+    @JsonProperty("hostAware")
+    public void setHostAware(final HostAware hostAware)
+    {
+        myHostAware = hostAware;
+    }
+
+    @JsonProperty("hostAware")
+    public HostAware getHostAware()
+    {
+        return myHostAware;
+    }
+
+    public enum ConnectionType
+    {
+        datacenterAware, rackAware, hostAware
+    }
+
+    public static final class DatacenterAware
+    {
+        private Map<String, Datacenter> myDatacenters = new HashMap<>();
+
+        public DatacenterAware()
+        {
+
+        }
+
+        @JsonProperty("datacenters")
+        public Map<String, Datacenter> getDatacenterAware()
+        {
+            return myDatacenters;
+        }
+
+        @JsonProperty("datacenters")
+        public void setDatacenters(final List<Datacenter> datacenters)
+        {
+            if (datacenters != null)
+            {
+                myDatacenters = datacenters.stream().collect(
+                        Collectors.toMap(Datacenter::getName, ks -> ks));
+            }
+        }
+
+        public static final class Datacenter
+        {
+            private String myName = "datacenter1";
```

Comment:
I'm not sure we should have a default name for datacenters, unless perhaps cassandra itself provides a similar default.

Reply:
Ok, I will search about it, if cassandra doesn't have a default value for it we can put as null or a empty string